### PR TITLE
Successfully recover after 2 reorgs where the second reorg failed

### PR DESCRIPTION
**Pull request by dangershony**
Writing tests to try to simulate this bug
https://github.com/stratisproject/StratisBitcoinFullNode/issues/3011

### CODE REVIEW

File: src/Stratis.Bitcoin.IntegrationTests/Program.cs [hunk 1]
```diff
@@ -12,6 +12,8 @@ internal class Program
     {
         public static void Main(string[] args)
         {
+            new ConsensusManagerFailedReorgTests().TwoConsecutive_ReorgChain_FailsFullValidation_Reconnect_OldChain_Nodes_Connected();
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/Program.cs]: Maybe just revert this ;)

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerFailedReorgTests.cs [hunk 2]
```diff
@@ -57,6 +60,78 @@ public void ReorgChain_FailsFullValidation_Reconnect_OldChain_Nodes_Connected()
...
+            using (var builder = NodeBuilder.Create(this))
+            {
+                var bitcoinNoValidationRulesNetwork = new BitcoinRegTestNoValidationRules();
+                var syncerNetwork = new ConsensusManagerTests.BitcoinOverrideRegTest();
+
+                var minerA = builder.CreateStratisPowNode(syncerNetwork).WithDummyWallet().Start();
+                var minerB = builder.CreateStratisPowNode(bitcoinNoValidationRulesNetwork).WithDummyWallet().Start();
+                var minerC = builder.CreateStratisPowNode(bitcoinNoValidationRulesNetwork).WithDummyWallet().Start();
+
+                // MinerA mines 5 blocks
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerFailedReorgTests.cs]: 15 blocks?

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerFailedReorgTests.cs [hunk 3]
```diff
@@ -57,6 +60,78 @@ public void ReorgChain_FailsFullValidation_Reconnect_OldChain_Nodes_Connected()
...
+                TestHelper.MineBlocks(minerA, 15);
+
+                // MinerB and MinerC syncs with MinerA
+                TestHelper.ConnectAndSync(minerA, minerB, minerC);
+                TestHelper.DisconnectAll(minerA, minerB, minerC);
+
+                // Miner A continues to mine to height 9
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerFailedReorgTests.cs]: Maybe just remove comment ?

File: src/Stratis.Bitcoin/Consensus/ConsensusManager.cs [hunk 4]
```diff
@@ -739,7 +739,7 @@ private async Task<ConnectBlocksResult> ReconnectOldChainAsync(List<ChainedHeade
 
             // We failed to jump back on the previous chain after a failed reorg.
             // And we failed to reconnect the old chain, database might be corrupted.
-            this.logger.LogError("A critical error has prevented reconnecting blocks");
+            this.logger.LogError("A critical error has prevented reconnecting blocks, error = {0}", connectBlockResult.Error);
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: Is LogError also logged with debug log level? To me, these critical errors should be in the normal node log.

**Comment by fassadlr** [comment]
@dangershony can you fix the conflicts?

**Comment by noescape00** [comment]
@dangershony please fix conflicts

**Comment by maciejzaleski** [comment]
@dangershony Could this PR be closed. I believe we have already resolved issue related to BIP30

**Comment by dangershony** [comment]
Yes I will clo

**Pull request closed without merging**